mindmeld/text_preparation/text_preparation_pipeline.py

```diff
@@ -335,6 +349,105 @@ def find_mindmeld_annotation_re_matches(text):
         """
         return list(MINDMELD_ANNOTATION_PATTERN.finditer(text))
 
+    @staticmethod
+    def calc_unann_spans(text):
+        """Calculates the spans of text that exclude mindmeld entity annotations.
+        For example, "{Lucien|person_name}" would return [(1,7)] since "Lucien" is
+        the only text that is not the annotation.
+
+        Args:
+            text (str): Original sentence with markup to modify.
+        Returns:
+            unann_spans (List[Tuple(int, int)]): The list of spans where each span
+                is a section of the original text excluding annotation markup. The
+                first element of the tuple is the start index and the second is the
+                ending index + 1.
+        """
+        matches = TextPreparationPipeline.find_mindmeld_annotation_re_matches(text)
+        unann_spans = []
+        prev_entity_end = 0
+        # MindMeld Entities are not present in text
+        if len(matches) == 0:
+            unann_spans = [(0, len(text))]
+            return unann_spans
+        for match in matches:
+            entity_start, entity_end = match.span()
+            entity_text = match.group(1)
+
+            unann_spans.append((prev_entity_end, entity_start))
+            entity_text_start = entity_start + 1
+            unann_spans.append(
+                (entity_text_start, entity_text_start + len(entity_text))
+            )
+            prev_entity_end = entity_end
+
+        if len(unann_spans) > 0:
+            unann_spans.append((prev_entity_end, len(text)))
+
+        # Filter out spans that have a length of 0
+        unann_spans = [span for span in unann_spans if span[1] - span[0] > 0]
+        return unann_spans
+
+    @staticmethod
+    def unann_to_ann_idx_map(unann_spans):
```

Comment:
nit: can we expand the unann to unannotated and similarly for ann? Improves readability I think.

Reply:
Sure, made the requested change